### ISSUE TITLE
Log4j2 configuration overrides are no longer loaded

### ISSUE THREAD

**Issue by lalmeras** [igloo-project/igloo-parent#72]
It seems it affects igloo 4.x+ versions. We need to check which versions are affected.

`log4j2-*.properties` are not loaded. Only `log4j2.properties` is loaded.

**Comment by lalmeras** [comment]
Behavior change comes with log4j2 2.19.x. log4j2 2.13.x - 2.18.x is OK. So only igloo 4.4+ is affected (including 5.x). We check this hypothesis.

**Comment by lalmeras** [comment]
Errata : Behavior change comes with log4j2 2.18.x. log4j2 2.13.x - 2.17.2 is OK. So only igloo 4.3+ is affected (including 5.x).

Overriding `org.apache.logging.log4j:*` dependencies with 2.17.2 version is a working fix for this issue.

**Comment by lalmeras** [comment]
log4j2 2.17.2 cannot be used with slf4j 2.x ; rollback to log4j2 2.17.2 implies slf4j rollback (no impact as slf4j 1.7 and 2.x are API-compatible).

**Comment by lalmeras** [comment]
For log4j2 2.18+, the problem is that `org.apache.logging.log4j.util.PropertiesUtil.Environment` previously has a `get(propertyName)` method that always lookup first system properties before using various property sources.

In igloo, we have a `log4j2.component.properties` with the content (automatically loaded by log4j2).

```ini
log4j2.configurationFile=log4j2.properties
```

So first load set `log4j2.configurationFile` in log4j2 configuration cache (cf Environment). Subsequent reload just loads again the same value from `log4j2.component.properties` and system property override is just ignored.

It explains this simple reproducer without Igloo does not exhibit the failure (if log4j2-custom.properties switch level to error, second message is not displayed with both 2.17.0 and 2.18.0):

```
		LogManager.getLogger().warn("start");
		System.setProperty("log4j.configurationFile", "log4j2.properties,log4j2-custom.properties");
		// reload log4j2 properties
		PropertiesUtil.getProperties().reload();
		// reload configuration
		((LoggerContext) LogManager.getContext(false)).reconfigure();
		LogManager.getLogger().warn("stop");
```

I need to check why this `log4j2.component.properties` was added.

**Comment by lalmeras** [comment]
35abb86156634f63534936ea9d77327c62a4e94f fixes issue with version 5.0.2.

**Comment by lalmeras** [comment]
Issue comes from a mixed usage of legacy `log4j.configurationFile` in programmatic configuration and `log4j2.configurationFile` in `log4j2.component.properties`. log4j2 2.17 and before seems to correctly override configuration even if a different property key is used. log4j2 2.18+ does not longer support this behavior.

**Comment by lalmeras** [comment]
b8197161afe1fe5af746cc7ffd857dd41217aafd fixes issue with version 4.4.2

01e3f155e1cc1365cf56a955355a315495c41381 fixes issue with version 4.3.1
